lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -3148,13 +3148,40 @@ TypeSystemSwiftTypeRef::GetTypeBitAlign(opaque_compiler_type_t type,
             AsMangledName(type));
   return {};
 }
+
 bool TypeSystemSwiftTypeRef::IsTypedefType(opaque_compiler_type_t type) {
-  return m_swift_ast_context->IsTypedefType(ReconstructType(type));
+  auto impl = [&]() {
+    using namespace swift::Demangle;
+    Demangler dem;
+    NodePointer node = GetDemangledType(dem, AsMangledName(type));
+    return node && node->getKind() == Node::Kind::TypeAlias;
+  };
+
+  VALIDATE_AND_RETURN(impl, IsTypedefType, type, (ReconstructType(type)),
+                      (ReconstructType(type)));
 }
+
 CompilerType
 TypeSystemSwiftTypeRef::GetTypedefedType(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetTypedefedType(ReconstructType(type));
+  auto impl = [&]() -> CompilerType {
+    using namespace swift::Demangle;
+    Demangler dem;
+    NodePointer node = GetDemangledType(dem, AsMangledName(type));
+    if (!node || node->getKind() != Node::Kind::TypeAlias)
+      return {};
+    auto pair = ResolveTypeAlias(m_swift_ast_context, dem, node);
+    if (NodePointer resolved = std::get<swift::Demangle::NodePointer>(pair)) {
```

Comment:
since this function has `using namespace swift::Demangle`, you don't need the fully qualified `<swift::Demangle::NodePointer>`, it could be reduced to `<NodePointer>`. Feel free to merge, not worth another round of testing to change just this.